File: hackerrank/easy/0014-sock-merchant/python/solution.py
## solution.py

from collections import Counter
from typing import List

def sock_merchant(n: int, ar: List[int]) -> int:
    counts = Counter(ar)
    return sum(v // 2 for v in counts.values())


# --- demo run ---
if __name__ == "__main__":
    arr = [10, 20, 20, 10, 10, 30, 50, 10, 20]
    print(sock_merchant(len(arr), arr))  # 3

File: hackerrank/easy/0014-sock-merchant/swift/solution.swift
// solution.swift

func sockMerchant(n: Int, ar: [Int]) -> Int {
    var counts: [Int: Int] = [:]
    for color in ar {
        counts[color, default: 0] += 1
    }
    return counts.values.reduce(0) { $0 + $1 / 2 }
}

// --- demo run ---
let arr = [10, 20, 20, 10, 10, 30, 50, 10, 20]
print(sockMerchant(n: arr.count, ar: arr))  // 3